### PR TITLE
Docs migration from XPK to CTK(3/5)

### PR DESCRIPTION
**Pull request by KevinYenky**
# Description

This PR is the third part of decomposing the XPK to Cluster Toolkit document migration (from PR #5175). It focuses on updating our post-training tutorial documentation for Reinforcement Learning to use Cluster Toolkit (CTK) instead of the deprecated XPK launcher.

**Specifically, this PR updates the following RL guides:**
- **Multi-Host RL (GRPO & GSPO):** Replaces `xpk workload create-pathways` with `gcluster job submit` for both GRPO and GSPO workloads, updating environment setup variables (like `COMPUTE_TYPE` and `TOPOLOGY`) and job monitoring commands.
- **Model-Specific RL Tutorials (Gemma 4, GPT-OSS 20B, Qwen3 30B):** Updates the step-by-step submission instructions to use CTK and `gcluster` commands instead of `xpk` scripts. 
- **RL README:** Updates prerequisite instructions to recommend Cluster Toolkit for standard GKE workloads over XPK.

**Why this is being made:**
CTK is replacing XPK as the primary submission mechanism for Google Kubernetes Engine (GKE) workloads. Breaking down the massive PR #5175 into smaller, logical chunks (starting with these core guides) makes the migration significantly easier to review and test.

# Tests
tested by scripts that run the gcluster command provided in the documents, results stored in gcs bucket.

# Checklist
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents.

### CODE REVIEW

File: docs/tutorials/posttraining/rl_gemma4_26b.md [hunk 1]
```diff
@@ -96,35 +105,72 @@ Both files are already included under `src/maxtext/examples/chat_templates/`, so
...
 
 ### Submit your workload
 
 ```bash
-# The Docker image you pushed in the previous step
-export DOCKER_IMAGE=<IMAGE_NAME>
-
-# Run the RL training script on your cluster
-run_tutorial maxtext/trainers/post_train/rl/scripts/run_gemma4_26b_rl.sh
+export COMPUTE_TYPE=<COMPUTE_TYPE>
```
Comment (SurbhiJainUSC) [inline docs/tutorials/posttraining/rl_gemma4_26b.md:113]: Can you instead update `maxtext/trainers/post_train/rl/scripts/run_gemma4_26b_rl.sh`? Same for other models.

Comment (KevinYenky) [reply docs/tutorials/posttraining/rl_gemma4_26b.md]: Done. Applied the changes to the 4 corresponding script files

File: src/maxtext/trainers/post_train/rl/scripts/run_gptoss_20b_rl.sh [hunk 2]
```diff
@@ -95,14 +96,14 @@ MAXTEXT_COMMAND="MODEL_IMPL_TYPE=flax_nnx \
...
   load_parameters_path=${MAXTEXT_CKPT_PATH}  \
-  run_name=${WORKLOAD_NAME}  \
+  run_name=${RUN_NAME}  \
   base_output_directory=${BASE_OUTPUT_DIRECTORY}  \
   checkpoint_storage_use_ocdbt=False \
   checkpoint_storage_use_zarr3=False \
   rollout_data_parallelism=2 \
   rollout_tensor_parallelism=8 \
   hbm_utilization_vllm=0.8 \
-  batch_size=8 profiler=xplane \
```
Comment (SurbhiJainUSC) [inline src/maxtext/trainers/post_train/rl/scripts/run_gptoss_20b_rl.sh:105]: Is this intentional to remove `batch_size`?

Comment (KevinYenky) [reply src/maxtext/trainers/post_train/rl/scripts/run_gptoss_20b_rl.sh:105]: Yes. This is because `batch_size` is already defined as 16 on line 111, so the one on this line was a duplicate and we only have to keep one.

**Review by SurbhiJainUSC — APPROVED** [review]

File: src/maxtext/trainers/post_train/rl/scripts/run_gemma4_26b_rl.sh [hunk 3]
```diff
@@ -162,17 +162,26 @@ vllm_hf_overrides='{architectures: [\"MaxTextForCausalLM\"]}' \
...
 
 # Workload Creation
-xpk workload create-pathways \
-  --cluster=$CLUSTER_NAME \
-  --project=$PROJECT_ID \
-  --zone=$ZONE \
-  --priority=medium \
-  --max-restarts=0 \
-  --tpu-type=$TPU_TYPE \
+gcloud config set project "$PROJECT_ID"
```
Comment (lydhr) [inline src/maxtext/trainers/post_train/rl/scripts/run_gemma4_26b_rl.sh:165]: These can stay in the pre-requisites as before.

File: src/maxtext/trainers/post_train/rl/scripts/run_gemma4_26b_rl.sh [hunk 4]
```diff
@@ -162,17 +162,26 @@ vllm_hf_overrides='{architectures: [\"MaxTextForCausalLM\"]}' \
...
+  --topology="${TOPOLOGY}" \
   --num-slices=1 \
-  --docker-image="${DOCKER_IMAGE}" \
-  --workload="${WORKLOAD_NAME}" \
-  --server-image="${PATHWAYS_SERVER_IMAGE}" \
-  --proxy-server-image="${PATHWAYS_PROXY_SERVER_IMAGE}" \
-  --custom-pathways-proxy-server-args='${XLA_FLAGS}' \
+  --priority=medium \
+  --restarts=0 \
+  --pathways-gcs-location="${BASE_OUTPUT_DIRECTORY}" \
```
Comment (lydhr) [inline src/maxtext/trainers/post_train/rl/scripts/run_gemma4_26b_rl.sh:182]: May I have any references about why do we need pathways-* flags here additionally?

File: src/maxtext/experimental/rl/README.md [hunk 5]
```diff
@@ -4,50 +4,53 @@ This directory contains code and documentation for **GRPO**, a reinforcement lea
...
-4.  **XPK:** (Recommended) Install and configure XPK, the Accelerated Processing Kit, to simplify the management of your GKE cluster and Pathways environment.
+1. **Google Cloud Platform (GCP):** You need a GCP project with sufficient TPU quota.
+2. **GKE Cluster with Pathways:**
+   - Follow the instructions to create a GKE cluster with Pathways support: [Create a GKE Cluster with Pathways](https://cloud.google.com/ai-hypercomputer/docs/workloads/pathways-on-cloud/create-gke-cluster)
+   - This involves setting up node pools for both training and inference.
+3. **Pathways Container Images:** Your GCP service account must be allowlisted to access the Pathways container images. Contact your Google Cloud representative for assistance.
+4. **Cluster Management Tools:** For standard GKE workloads, [Cluster Toolkit](../../../../docs/run_maxtext/run_maxtext_via_cluster_toolkit.md) is recommended. For legacy Pathways-specific orchestration, install and configure XPK to manage your GKE cluster and Pathways environment.
```
Comment (lydhr) [inline src/maxtext/experimental/rl/README.md:23]: To confirm, does this mean we are not running pathways in RL? "For legacy Pathways-specific orchestration, install and configure XPK to manage"

File: docs/tutorials/posttraining/rl_qwen3_30b.md [hunk 6]
```diff
@@ -26,8 +26,8 @@ Before starting, ensure you have:
 - Access to a Google Cloud Project with TPU quotas.
 - A Hugging Face account with an access token for downloading models.
 - Permissions for Google Artifact Registry (Artifact Registry Writer role).
-- Prerequisites for XPK installed (follow [official documentation](https://github.com/AI-Hypercomputer/xpk/blob/main/docs/installation.md#1-prerequisites)).
-- A Pathways-ready GKE cluster (see [create GKE cluster](https://docs.cloud.google.com/ai-hypercomputer/docs/workloads/pathways-on-cloud/create-gke-cluster)).
```
Comment (lydhr) [inline docs/tutorials/posttraining/rl_qwen3_30b.md:30]: GKE cluster creation can be the same.
Or `Use Cluster Toolkit to deploy a pathways enabled GKE cluster if needed:
gcluster deploy gke-tpu-v7x.yaml` [Reference](https://docs.google.com/document/d/1h2PAIPFFcFj7nXudGOCR-bGJJTsNxliI4IW6U3om6Qw/edit?resourcekey=0-Q-1e5WoCavLmrrtgkU2NeQ&tab=t.0)

File: docs/tutorials/posttraining/rl_qwen3_30b.md [hunk 7]
```diff
@@ -84,7 +87,7 @@ export MAXTEXT_CKPT_PATH=<CKPT_PATH> # e.g., gs://my-bucket/my-model-checkpoint/
 
 ### Build and Upload MaxText Docker Image
 
-For instructions on building and uploading the MaxText Docker image with post-training dependencies, please refer to the [official documentation](build-docker).
```
Comment (lydhr) [inline docs/tutorials/posttraining/rl_qwen3_30b.md:87]: The anchor was not changed.

**Review by lydhr — CHANGES_REQUESTED** [review]

File: docs/tutorials/posttraining/rl_gemma4_26b.md [hunk 8]
```diff
@@ -108,23 +111,30 @@ export DOCKER_IMAGE=<IMAGE_NAME>
...
 
-# List pods to find the specific name
-kubectl get pods | grep ${WORKLOAD_NAME}
+# List pods (use jobset-name to select both head and worker pods in Pathways)
+kubectl get pods -l jobset.sigs.k8s.io/jobset-name=${RUN_NAME?}
 
-# stream the logs from the running pod (replace <POD_NAME> with the name you found)
-kubectl logs -f <POD_NAME>
+# Stream logs with kubectl
+kubectl logs -f -l jobset.sigs.k8s.io/jobset-name=${RUN_NAME?} --all-containers=true
```
Comment (darisoy) [inline docs/tutorials/posttraining/rl_gemma4_26b.md]: Will this work on the tutorial's slice? `kubectl logs -f` with a selector follows at most 5 streams by default ([logs.go](https://github.com/kubernetes/kubernetes/blob/f139c507930ad25a5118dc44d284d8212384fd55/staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go#L385-L389)). With `--all-containers=true`, every container counts as a stream, and a 4x4x4 v7x slice has 16 worker pods plus a 3-container head pod. Should we add `--max-log-requests=<N>`, or point to `gcluster job logs --main-only=false` instead? The same line appears in the other three RL tutorials.

**Review by darisoy — COMMENTED** [review]
Thanks for this! Nice catch on `rl.loss_algo=gspo-token`. Since `loss_algo` lives under `rl:` in `rl.yml`, I think the old top-level override was being silently ignored. As in [#5176](https://github.com/AI-Hypercomputer/maxtext/pull/5176#pullrequestreview-5310660980), should we keep `ENABLE_PATHWAYS_PERSISTENCE=1` on the `--pathways` commands in `rl_on_multi_host.md`? The `run_*_rl.sh` scripts still set it, so otherwise the tutorial and the scripts would checkpoint differently.
